### DISCUSSION TITLE
How to change predicted instances to user-labeled instances for training without double-clicking each one in GUI?

### DISCUSSION THREAD

**Discussion by amblypatty** [talmolab/sleap#1731] [Help!]
I have an slp file that contains updated instances from multiple individual-video slps that were updated using code from [this discussion](https://github.com/talmolab/sleap/discussions/1717) and merged into one slp file. However, the updated instances have the property of `PredictedInstance`s, where I instead would like to consider them as `user_instances` so they can all be input into the training data. The number of `predicted_instances` frames is too large to go through manually in the GUI and double-click each one. Is there a way to use the CLI or ipython to switch this property?

I am using SLEAP v1.3.3 and use Google Colab for training and inference.

**Comment by Marielamend** [comment]
Hello @amblypatty!

Other than double-clicking each predicted instance, you can also go to **Labels** > **Add Instances from All Predictions on Current Frame** and it will convert to a regular instance but only for the current frame.
![image](https://github.com/talmolab/sleap/assets/120222245/2088d8f5-cc38-4120-b5a9-703acecb999e)

Considering the workflow that sleap is made for --> entering a human-in-the-loop iterations where you use the predictions to reach a desired pose labeling accuracy, we do not want every single frame to be a user-labeled frame so there is no feature for this on the GUI. 

If you want move forward with doing so, a great resource is this [data structures](https://sleap.ai/notebooks/Data_structures.html) notebook that you can reference to do so!

let us know if you need some assistance with this!
-Mariela:)

**Reply by amblypatty** [reply]
Hi @Marielamend,

Thank you for the reply. I have been using the human-in-the-loop method to correct instances for the last year on this one project, however, in order to fix multiple sections where I have been seeing inference problems after training, I have come to the conclusion that I just need drastically more training labels to attempt training with. I needed to speed some things up for increasing the number of training labels I have, so I exported my data, deleted erroneous sections and used linear interpolation on those sections where appropriate. The linear interpolation workflow has drastically increased the number of labeled frames that I, the user, would approve to be useful for training, but there are thousands of manually and linear-interpolated instance frames. Utilizing the notebook from the discussion referenced above and sleap-io, I was able to reimport the corrected labels, but they were imported as `PredictedInstance`s rather than `user_instances`. 

Since asking this question, I have been working with the Data Structures notebook and have been attempting to add `Instance`s to each frame with `user-labeled` attributes that are copies of the current `PredictedInstance`s and then delete the predicted instances and resave the slp file. However, I am continually running into errors that are difficult to figure out when looking at the sleap source code, but I would like to not have to submit a new question every time I run into a new error. If you could be of assistance with this, that would be wonderful, just let me know what else you need from me.

-Patrick
